src/main/webapp/filter_tag_area.js

```diff
@@ -0,0 +1,143 @@
+      }
+    });
+
+    /* This button does not move- when clicked it opens a filter entry area */
+    this.addFilterButton = document.createElement("div");
+    this.addFilterButton.setAttribute("class", "filter-tag-area");
+    this.addFilterButton.setAttribute("id", "add-filter-button");
+    this.addFilterButton.textContent = "+ Add Filter";
+    this.addFilterButton.addEventListener('click', () => {
+      /* If the user is on the second part of entering a filter, add filter is disabled */
+      if (!this.filterEntryArea.contains(this.filterParamInput)) {
+        this.filterTypeInput.appendChild(this.filterDataList);
+        this.filterEntryArea.appendChild(this.filterEntryClose);
+        this.filterEntryArea.appendChild(this.filterTypeInput);
+        this.activeFilterArea.appendChild(this.filterEntryArea);
+      }
+    });
+    this.activeFilterArea.appendChild(this.addFilterButton);
+
+    this.parentSearchArea.organizationSearchArea.appendChild(this.activeFilterArea);
+  }
+
+  addFilterTag(urlParamKey, urlParamValue) {
+    let filterTagArea = document.createElement("div");
+    filterTagArea.setAttribute("class", "filter-tag-area");
+    /* ID is given to zipcode tag so it can be removed if new one is added */
+    if (urlParamKey === "zipcode") {
+      filterTagArea.setAttribute("id", "zipcodeTag");
+    }
+ 
+    let filterTagLabel = document.createElement("div");
+    filterTagLabel.textContent = urlParamValue;
+    filterTagLabel.setAttribute("class", "filter-tag-label");
+    filterTagArea.appendChild(filterTagLabel);
+ 
+    let filterTagClose = document.createElement("div");
+    filterTagClose.addEventListener('click', () => this.removeFilterTag(urlParamKey, urlParamValue, filterTagArea));
+    filterTagClose.textContent = 'X';
+    filterTagClose.setAttribute("class", "filter-tag-close");
+    filterTagArea.appendChild(filterTagClose);
+
+    this.activeFilterArea.appendChild(filterTagArea);
+  }
+
+  async removeFilterTag(urlParamKey, urlParamValue, filterTag) {
+    if (urlParamKey === "zipcode") {
+      this.parentSearchArea.filterParams.delete("zipcode");
+    } else {
+      if (this.parentSearchArea.filterParams.getAll(urlParamKey).length === 1) {
+        /* If only 1 filter param, delete the array */
+        this.parentSearchArea.filterParams.delete(urlParamKey);
+      } else {
+        /* If not, just remove specified element */
+        let filterArray = this.parentSearchArea.filterParams.getAll(urlParamKey);
+        filterArray.splice(filterArray.indexOf(urlParamValue), 1);
+        this.parentSearchArea.filterParams.set(urlParamKey, filterArray);
+      }
+    }
+    this.activeFilterArea.removeChild(filterTag);
+    this.parentSearchArea.organizationObjectsList = [];
+    this.parentSearchArea.organizationListArea.innerHTML = "";
+    await this.parentSearchArea.getListOfOrganizations();
+    this.parentSearchArea.renderListOfOrganizations();
```

Comment:
Why update the org list here, but not in addFilterTag (+elsewhere)?

Reply:
removeFilterTag is the function that is called when the "x" on a filter tag is clicked, so it clears the organization list and then re-queries for it with the new set of filters. addFilterTag doesn't need to worry about updating the org list because it is only designed to add the tag to the dom, since when new url params are added getting the new list is handled by the url search params function in search_area.js

Reply:
Rephrased: updating the org list and removing a filter seem like unrelated tasks, tied together only by the context in which they're used.

Can you break these apart into their own functions? (And reuse the refreshOrgList function wherever applicable!)

Reply:
I was originally updating the org list in the setUrlParamValue function after addFilterTag was called, but I agree that it makes it the code a little bit more readable to have addFilterTag and removeFilterTag do the same thing, so the new commit changes it to do it that way